include/nuSQuIDS/body.h

```diff
@@ -762,6 +766,123 @@ class EarthAtm: public Body{
+    /// \brief vector containing interpolators for the differential flux
+    std::vector<TriCubicInterpolator> Interpolators;
+    /// \brief indices for the electron and muon flavors
+    /// defaulted to the first and second flavor states
+    int E_nu_index = 0;
+    int Mu_nu_index = 1;
+    
+    bool init_emit_energies = false;
+    
+  public:
+  
+    /// \brief Default constructor using supplied MCEQ neutrino flux.
+    EmittingEarthAtm();
+    
+    /// \brief constructor with user passed energy space
+    EmittingEarthAtm(marray<double, 1> ESpace);
+    
+    /// \brief Constructor from a user supplied production model.
+    /// @param prodmodel Path to the production model data file.    
+    /// \details The input file should have five columns.
+    /// The first one must run in descending order from or between 1 and -1
+    /// representing the range of coszen values corresponding to tracks
+    /// relevant to your propagation
+    /// The second column must contain the height within the atmosphere in cm
+    /// in descending order from or between 6000000 (60 km) and 0
+    /// The 3rd column must correspond to the energy in GeV 
+    /// in descending order over any relevant range
+    /// The 4th and 5th columns must correspond to the differential E_nu and Mu_nu
+    /// flux corresponding to the coszen, height, and energy in the row
+    EmittingEarthAtm(std::string prodmodel);
+    
+    /// \brief Deconstructor
+    ~EmittingEarthAtm();
+    
+    /// \brief Function that passes the user's list of energies to the class
+    /// \details Used by nuSQUIDSATM implementation to provide the 
+    /// class with the energy list required for retrieval of interpolation
+    void Set_EmissionEnergies(const marray<double,1>& ESpace_){
+      ESpace = ESpace_;
+      init_emit_energies = true;
+    }
+    /// \brief Returns true if neutrino flux emission from bodies is considered
+    bool Get_EnergyInit(){
+      return init_emit_energies;
+    }
+    /// \brief Function that sets the height of the atmosphere
+    /// \details Used by nuSQUIDSATM to set the height for every
+    /// instance of the body
+    void SetAtmosphereHeight(double height){
+      atm_height = height;
+      earth_with_atm_radius = radius + atm_height;
+    }
+    /// \brief set index of produced flavors
+    /// \details Set the flavor index of the electron and muon neutrinos
+    /// or to different flavor indices as user input data requires
+    void Set_ProducedFlavors(int E_nu_index_, int Mu_nu_index_){
+      E_nu_index = E_nu_index_;
+      Mu_nu_index = Mu_nu_index_;
+    }
+   
+    //Important function that overrides the default flux injection with ATM data
+    void injected_neutrino_flux(marray<double, 3>& flux, const GenericTrack& track, const nuSQUIDS& nusquids) override;
+    
+    //Helpful function for data parsing
+    void unique_sort(std::vector<double>& vec, double& min_val, double& max_val, long unsigned int& number_val);
```

Comment:
This doesn't appear to need to be either a member function or a part of the public interface; it could just be a free function in the implementation file.